packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -317,13 +317,22 @@ class Auth {
         const refreshToken = refreshTokenRegistered || refreshTokenGuest
         if (refreshToken) {
             try {
-                return this.queueRequest(
+                return await this.queueRequest(
                     () => helpers.refreshAccessToken(this.client, {refreshToken}),
                     !!refreshTokenGuest
                 )
-            } catch {
-                // If anything bad happens during refresh token flow
-                // we continue with the PKCE guest user flow.
+            } catch (error) {
+                // If the refresh token is invalid, we need to re-login the user
+                if (error instanceof Error && 'response' in error) {
+                    // commerce-sdk-isomorphic throws a `ResponseError`, but doesn't export the class.
+                    // We can't use `instanceof`, so instead we just check for the `response` property
+                    // and assume it is a fetch Response.
+                    const json = await (error['response'] as Response).json()
+                    if (json.message === 'invalid refresh_token') {
```

Comment:
There's not a lot of context for this change in the PR description.
I'm assuming we need to re-implement this because it's a functionality that got lost during the transition from `commerce-api` folder to `commerce-sdk-react`.

If that is the case, Do we want to also address the expired token scenario as we did in the `commerce-api` folder?
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/37c932ec02d83fe7961ed1b9cb8717eea9952434/packages/template-retail-react-app/app/commerce-api/auth.js#L214

Reply:
What we did in `commerce-api` was to consider either for access_token and refresh_token. We handled that slightly different in commerce-react auth, we have checked for access_token expiration first, if it is still available, we don't need to check for refresh_token. In this PR, we only need to fix the bug when refresh_token is invalid